xcode/Subconscious/Shared/Components/AddressBook/AddressBookModel.swift

```diff
@@ -61,25 +166,93 @@ struct AddressBookModel: ModelProtocol {
                 }
             }
             
-            return Update(state: model)
+            let fx: Fx<AddressBookAction> =
+                environment.data.addressBook.listEntries()
+                .map({ follows in
+                    AddressBookAction.populate(follows)
+                })
+                .eraseToAnyPublisher()
             
-        case .follow(did: let did, petname: let petname):
+            return Update(state: model, fx: fx)
+            
+        case .requestFollow:
+            var model = state
+            // Show errors on any untouched fields, hints at why you cannot submit
+            model.followUserForm.did = FormField.update(
```

Comment:
Same as above. Should be:

```swift
return update(
    state: model,
    actions: [
        .didField(.markAsTouched),
        .petnameField(.markAsTouched)
    ],
    environment: environment
)
```

Reply:
Refactored 👍 

I introduced a new action `.attemptFollow` with `update` as an intermediate step to clean up the logic. So it flows from:

`.requestFollow` -> `.attemptFollow` then either `.succeedFollow` or `.failFollow`